### PR TITLE
[5.8] Fix session issue when session.path is not root

### PR DESCRIPTION
**Pull request by matrad**
If we set the session path to a subdirectory, for example ```/backend``` then if we go to a page outside the ```/backend``` path-segment then the already initiated session from a previous route is not sent back to laravel, and laravel sends a new session-cookie, so the previous session will be overwritten.

With this fix the session only will be started, if we are within the path segment of ```session.path```.

### CODE REVIEW

File: src/Illuminate/Session/Middleware/StartSession.php [hunk 1]
```diff
@@ -44,6 +45,12 @@ public function handle($request, Closure $next)
             return $next($request);
         }
 
+        $sessionCookiePath = config('session.path', '/');
```
Comment (X-Coder264) [inline src/Illuminate/Session/Middleware/StartSession.php]: `$this->manager->getSessionConfig()` is used multiple times inside this middleware to fetch the session config, I think we should be consistent.

**Comment by X-Coder264** [comment]
I doubt that changes like these to session management are gonna be merged for a patch release, either way IMHO a test is needed for a change like this in such an important part of the framework.

File: src/Illuminate/Session/Middleware/StartSession.php (hunk 1 above)
Comment (matrad) [reply src/Illuminate/Session/Middleware/StartSession.php]: @X-Coder264 fixed.

File: src/Illuminate/Session/Middleware/StartSession.php [hunk 2]
```diff
@@ -44,6 +45,12 @@ public function handle($request, Closure $next)
             return $next($request);
         }
 
+        $sessionCookiePath = $this->manager->getSessionConfig()['path'];
```
Comment (GrahamCampbell) [inline src/Illuminate/Session/Middleware/StartSession.php]: This will crash if 'path' is not set?

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

**Comment by taylorotwell** [comment]
I am closing this pull request because it lacks sufficient explanation, tests, or both. It is difficult for us to merge pull requests without these things because the change may introduce breaking changes to the framework.

Feel free to re-submit your change with a **thorough** explanation of the feature and tests - integration tests are preferred over unit tests. Please include it's benefit to end users; the reasons it does not break any existing features; how it makes building web applications easier, etc.

Thanks!

**Pull request closed without merging**